Fix error handling when adding neighbor that does not support gossip

```
panic: peer does not support gossip

goroutine 401 [running]:
github.com/iotaledger/goshimmer/packages/gossip.GetAddress(0xc00006e200, 0x1575940, 0xc0004ee058)
        /root/goshimmer/packages/gossip/common.go:17 +0xa4
github.com/iotaledger/goshimmer/plugins/gossip.configureEvents.func4(0xc00006e200)
        /root/goshimmer/plugins/gossip/plugin.go:57 +0x4f
github.com/iotaledger/goshimmer/packages/gossip.peerCaller(0xba3740, 0xd15520, 0xc0004ae630, 0x1, 0x1)
        /root/goshimmer/packages/gossip/events.go:31 +0x62
github.com/iotaledger/hive.go/events.(*Event).Trigger(0xc000118a50, 0xc0004ae630, 0x1, 0x1)
        /root/go/pkg/mod/github.com/iotaledger/hive.go@v0.0.0-20200110132858-ea86cdb9d91e/events/event.go:31 +0xdb
github.com/iotaledger/goshimmer/packages/gossip.(*Manager).addNeighbor(0xc00078c900, 0xc00006e200, 0xc000636ec0, 0x0, 0x0)
        /root/goshimmer/packages/gossip/manager.go:183 +0xd8
github.com/iotaledger/goshimmer/packages/gossip.(*Manager).AddInbound(0xc00078c900, 0xc00006e200, 0xd15260, 0x0)
        /root/goshimmer/packages/gossip/manager.go:102 +0x8e
github.com/iotaledger/goshimmer/plugins/gossip.configureEvents.func2.1(0xc0000f1fb0)
        /root/goshimmer/plugins/gossip/plugin.go:43 +0x50
created by github.com/iotaledger/goshimmer/plugins/gossip.configureEvents.func2
        /root/goshimmer/plugins/gossip/plugin.go:42 +0x3f
root@goshimmer:~/goshimmer# vim config.json

```

--- a/packages/autopeering/peer/local.go
+++ b/packages/autopeering/peer/local.go
@@ -50,18 +50,9 @@ func NewLocal(network string, address string, db DB) (*Local, error) {
 	if l := len(key); l != ed25519.PrivateKeySize {
 		return nil, fmt.Errorf("invalid key length: %d, need %d", l, ed25519.PublicKeySize)
 	}
-	services, err := db.LocalServices()
-	if err != nil {
-		return nil, err
-	}
-	serviceRecord := services.CreateRecord()
-
-	// update the external address used for the peering and store back in DB
+	// update the external address used for the peering
+	serviceRecord := service.New()
 	serviceRecord.Update(service.PeeringKey, network, address)
-	err = db.UpdateLocalServices(serviceRecord)
-	if err != nil {
-		return nil, err
-	}
 
 	return newLocal(key, serviceRecord, db), nil
 }
@@ -81,12 +72,8 @@ func (l *Local) UpdateService(key service.Key, network string, address string) e
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
-	// update the service in the read protected map and store back in DB
+	// update the service in the read protected map
 	l.serviceRecord.Update(key, network, address)
-	err := l.db.UpdateLocalServices(l.serviceRecord)
-	if err != nil {
-		return err
-	}
 
 	// create a new peer with the corresponding services
 	l.Peer = *NewPeer(l.key.Public(), l.serviceRecord)

--- a/packages/autopeering/peer/mapdb.go
+++ b/packages/autopeering/peer/mapdb.go
@@ -4,16 +4,14 @@ import (
 	"sync"
 	"time"
 
-	"github.com/iotaledger/goshimmer/packages/autopeering/peer/service"
 	"github.com/iotaledger/hive.go/logger"
 )
 
 // mapDB is a simple implementation of DB using a map.
 type mapDB struct {
-	mutex    sync.RWMutex
-	m        map[string]peerEntry
-	key      PrivateKey
-	services *service.Record
+	mutex sync.RWMutex
+	m     map[string]peerEntry
+	key   PrivateKey
 
 	log *logger.Logger
 
@@ -34,10 +32,9 @@ type peerPropEntry struct {
 // NewMemoryDB creates a new DB that uses a GO map.
 func NewMemoryDB(log *logger.Logger) DB {
 	db := &mapDB{
-		m:        make(map[string]peerEntry),
-		services: service.New(),
-		log:      log,
-		closing:  make(chan struct{}),
+		m:       make(map[string]peerEntry),
+		log:     log,
+		closing: make(chan struct{}),
 	}
 
 	// start the expirer routine
@@ -68,24 +65,6 @@ func (db *mapDB) LocalPrivateKey() (PrivateKey, error) {
 	}
 
 	return db.key, nil
-}
-
-// LocalServices returns the services stored in the database or creates an empty map.
-func (db *mapDB) LocalServices() (service.Service, error) {
-	db.mutex.RLock()
-	defer db.mutex.RUnlock()
-
-	return db.services, nil
-}
-
-// UpdateLocalServices updates the services stored in the database.
-func (db *mapDB) UpdateLocalServices(services service.Service) error {
-	record := services.CreateRecord()
-
-	db.mutex.Lock()
-	defer db.mutex.Unlock()
-	db.services = record
-	return nil
 }
 
 // LastPing returns that property for the given peer ID and address.

--- a/packages/autopeering/peer/peerdb.go
+++ b/packages/autopeering/peer/peerdb.go
@@ -7,7 +7,6 @@ import (
 	"sync"
 	"time"
 
-	"github.com/iotaledger/goshimmer/packages/autopeering/peer/service"
 	"github.com/iotaledger/goshimmer/packages/database"
 	"github.com/iotaledger/hive.go/logger"
 )
@@ -29,10 +28,6 @@ const (
 type DB interface {
 	// LocalPrivateKey returns the private key stored in the database or creates a new one.
 	LocalPrivateKey() (PrivateKey, error)
-	// LocalServices returns the services stored in the database or creates an empty services.
-	LocalServices() (service.Service, error)
-	// UpdateLocalServices updates the services stored in the database.
-	UpdateLocalServices(services service.Service) error
 
 	// Peer retrieves a peer from the database.
 	Peer(id ID) *Peer
@@ -72,8 +67,7 @@ const (
 	dbNodePong = "lastpong"
 
 	// Local information is keyed by ID only. Use localFieldKey to create those keys.
-	dbLocalKey      = "key"
-	dbLocalServices = "services"
+	dbLocalKey = "key"
 )
 
 // NewPersistentDB creates a new persistent DB.
@@ -190,33 +184,6 @@ func (db *persistentDB) LocalPrivateKey() (PrivateKey, error) {
 	}
 
 	return key, nil
-}
-
-// LocalServices returns the services stored in the database or creates an empty services.
-func (db *persistentDB) LocalServices() (service.Service, error) {
-	key, err := db.db.Get(localFieldKey(dbLocalServices))
-	if err == database.ErrKeyNotFound {
-		return service.New(), nil
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	services, err := service.Unmarshal(key)
-	if err != nil {
-		return nil, err
-	}
-
-	return services, nil
-}
-
-// UpdateLocalServices updates the services stored in the database.
-func (db *persistentDB) UpdateLocalServices(services service.Service) error {
-	value, err := services.CreateRecord().CreateRecord().Marshal()
-	if err != nil {
-		return err
-	}
-	return db.db.Set(localFieldKey(dbLocalServices), value)
 }
 
 // LastPing returns that property for the given peer ID and address.

--- a/plugins/analysis/client/plugin.go
+++ b/plugins/analysis/client/plugin.go
@@ -94,16 +94,20 @@ func setupHooks(plugin *node.Plugin, conn *network.ManagedConnection, eventDispa
 		eventDispatchers.RemoveNode(ev.Peer.ID().Bytes())
 	})
 
+	onAddChosenNeighbor := events.NewClosure(func(ev *selection.PeeringEvent) {
+		if ev.Status {
+			eventDispatchers.ConnectNodes(local.GetInstance().ID().Bytes(), ev.Peer.ID().Bytes())
+		}
+	})
+
 	onAddAcceptedNeighbor := events.NewClosure(func(ev *selection.PeeringEvent) {
-		eventDispatchers.ConnectNodes(ev.Peer.ID().Bytes(), local.GetInstance().ID().Bytes())
+		if ev.Status {
+			eventDispatchers.ConnectNodes(ev.Peer.ID().Bytes(), local.GetInstance().ID().Bytes())
+		}
 	})
 
 	onRemoveNeighbor := events.NewClosure(func(ev *selection.DroppedEvent) {
 		eventDispatchers.DisconnectNodes(ev.DroppedID.Bytes(), local.GetInstance().ID().Bytes())
-	})
-
-	onAddChosenNeighbor := events.NewClosure(func(ev *selection.PeeringEvent) {
-		eventDispatchers.ConnectNodes(local.GetInstance().ID().Bytes(), ev.Peer.ID().Bytes())
 	})
 
 	// setup hooks /////////////////////////////////////////////////////////////////////////////////////////////////////

--- a/plugins/autopeering/plugin.go
+++ b/plugins/autopeering/plugin.go
@@ -51,10 +51,14 @@ func configureEvents() {
 		log.Infof("Salt updated; expires=%s", ev.Public.GetExpiration().Format(time.RFC822))
 	}))
 	selection.Events.OutgoingPeering.Attach(events.NewClosure(func(ev *selection.PeeringEvent) {
-		log.Infof("Peering chosen: %s / %s", ev.Peer.Address(), ev.Peer.ID())
+		if ev.Status {
+			log.Infof("Peering chosen: %s / %s", ev.Peer.Address(), ev.Peer.ID())
+		}
 	}))
 	selection.Events.IncomingPeering.Attach(events.NewClosure(func(ev *selection.PeeringEvent) {
-		log.Infof("Peering accepted: %s / %s", ev.Peer.Address(), ev.Peer.ID())
+		if ev.Status {
+			log.Infof("Peering accepted: %s / %s", ev.Peer.Address(), ev.Peer.ID())
+		}
 	}))
 	selection.Events.Dropped.Attach(events.NewClosure(func(ev *selection.DroppedEvent) {
 		log.Infof("Peering dropped: %s", ev.DroppedID)

--- a/plugins/gossip/plugin.go
+++ b/plugins/gossip/plugin.go
@@ -39,6 +39,9 @@ func configureEvents() {
 		}()
 	}))
 	selection.Events.IncomingPeering.Attach(events.NewClosure(func(ev *selection.PeeringEvent) {
+		if !ev.Status {
+			return // ignore rejected peering
+		}
 		go func() {
 			if err := mgr.AddInbound(ev.Peer); err != nil {
 				log.Debugw("error adding inbound", "id", ev.Peer.ID(), "err", err)
@@ -46,6 +49,9 @@ func configureEvents() {
 		}()
 	}))
 	selection.Events.OutgoingPeering.Attach(events.NewClosure(func(ev *selection.PeeringEvent) {
+		if !ev.Status {
+			return // ignore rejected peering
+		}
 		go func() {
 			if err := mgr.AddOutbound(ev.Peer); err != nil {
 				log.Debugw("error adding outbound", "id", ev.Peer.ID(), "err", err)

